sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -48,6 +49,22 @@ case class Mode(
 
   override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
 
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val defaultCheck = super.checkInputDataTypes()
+    if (defaultCheck.isFailure) {
+      defaultCheck
+    } else {
+      child.dataType match {
+        case _: StructType | _: ArrayType if !UnsafeRowUtils.isBinaryStable(child.dataType) =>
+          TypeCheckResult.TypeCheckFailure(
+            s"Input to function mode was a complex type" +
+              s" with non-binary collated fields," +
+              s" which is not yet supported by mode.")
+        case _ => TypeCheckResult.TypeCheckSuccess
+      }
```

Comment:
```suggestion
      if (UnsafeRowUtils.isBinaryStable(child.dataType) || child.dataType.isInstanceOf[StringType]) {
        TypeCheckResult.TypeCheckSuccess
      } else {
        ...
      }
```

Reply:
let's try to white-list instead of black-list (if we can, of course)

Reply:
@uros-db done